src/inmanta/agent/handler.py

```diff
@@ -394,43 +395,20 @@ def update_changes(  # noqa: F811
     def changes(self) -> Dict[str, AttributeStateChange]:
         return self._changes
 
-    def log_msg(self, level: int, msg: str, args: abc.Sequence[object], kwargs: abc.Mapping[str, object]) -> None:
-        LOGGER.warning(
-            "Direct calls to the log_msg method are being deprecated, please use the IBasicLogging interface instead."
-        )
-        if len(args) > 0:
-            raise Exception("Args not supported")
+    def log_msg(self, level: int, msg: str, args: abc.Sequence[object], kwargs: abc.MutableMapping[str, object]) -> None:
+        LOGGER.warning("Direct calls to the log_msg method are being deprecated, please use the LoggerABC interface instead.")
         if "exc_info" in kwargs:
-            exc_info = kwargs["exc_info"]
+            exc_info = bool(kwargs["exc_info"])
             kwargs["traceback"] = traceback.format_exc()
         else:
             exc_info = False
+        self._log_msg(level, msg, *args, exc_info=exc_info, **kwargs)
```

Comment:
You're passing it twice now. I don't think it would cause issues but I think it's cleaner to just go with this.
```suggestion
        self._log_msg(level, msg, *args, **kwargs)
```

This also means you can get rid of the `exc_info` variable above, and the traceback part is also redundant because that is done in the `_log_msg` method.